src/pages/[course_name]/chat.tsx

```diff
@@ -42,35 +43,41 @@ const ChatPage: NextPage = () => {
   useEffect(() => {
     if (!courseName && curr_route_path != '/gpt4') return
     const courseMetadata = async () => {
-      setIsLoading(true) // Set loading to true before fetching data
+      setIsLoading(true)
 
       // Handle /gpt4 page (special non-course page)
       let curr_course_name = courseName
       if (courseName == '/gpt4') {
         curr_course_name = 'gpt4'
       }
 
-      const response = await fetch(
-        `/api/UIUC-api/getCourseMetadata?course_name=${curr_course_name}`,
-      )
-      const data = await response.json()
+      // Fetch course metadata and document count in parallel
+      const [metadataResponse, documentsResponse] = await Promise.all([
+        fetch(`/api/UIUC-api/getCourseMetadata?course_name=${curr_course_name}`),
+        fetch(`/api/materialsTable/fetchProjectMaterials?from=0&to=0&course_name=${curr_course_name}`)
```

Comment:
This could really slow down our most critical page load speed on /chat. We might want to do this in the background instead of it blocking the page load. 

getCourseMetadata is Redis and returns in ~1 ms
fetchProjectMaterials is Postgres and returns in ~100-250 ms.